### PR TITLE
Feature/27 get wallets list by wallet name

### PR DESCRIPTION
**Pull request by Krisu93**

### CODE REVIEW

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/repository/WalletRepository.java [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
 
+import java.util.List;
+
 /**
  * Repository for Wallet Entity.
  * @Since 0.2.0
  */
 public interface WalletRepository extends JpaRepository<Wallet, Long> {
+    @Query("SELECT w FROM Wallet w WHERE w.name LIKE %:name%")
+    List<Wallet> findAllByLikeName(@Param("name") String name);
```
Comment (DawidStuzynski) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/repository/WalletRepository.java:16]: You don't have to create your own SQL queries, you can just use query methods provided by Spring JPA.
Here is a referecne: https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#appendix.query.method.subject

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/service/WalletServiceImpl.java [hunk 2]
```diff
@@ -82,7 +83,13 @@ public WalletDTO findById(Long id) {
     }
 
     @Override
-    public List<WalletDTO> findByName(String name) {
-        return getAll().stream().filter(walletDTO -> walletDTO.name().contains(name)).toList();
+    public List<WalletDTO> findAllByName(String name) {
+        List<WalletDTO> walletsDTO;
```
Comment (DawidStuzynski) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/service/WalletServiceImpl.java:87]: I think that listOfWalletDTO would be more descriptive.

File: src/main/resources/db/changelog/001-create-table-wallets.yaml [hunk 3]
```diff
@@ -23,15 +23,16 @@ databaseChangeLog:
...
                        primaryKey: true
                        nullable: false
                 -  column:
-                     name: firstname
+                     name: name
                      type: varchar(50)
                 -  column:
-                     name: lastname
-                     type: varchar(50)
+                     name: creation_date
```
Comment (DawidStuzynski) [inline src/main/resources/db/changelog/001-create-table-wallets.yaml:34]: One column cannot have 2 names and 2 types, I think that is fault caused by merging

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/WalletControllerTest.java [hunk 4]
```diff
@@ -436,7 +436,7 @@ void shouldReturnListOfAllWalletsByName() throws Exception {
         List<WalletDTO> listOfWalletsDTO = createListOfWalletsDTO();
         List<WalletDTO> foundedWalletsDTO = List.of(new WalletDTO(ID_OF_WALLET_2, NAME_OF_WALLET_2, CREATION_DATE_OF_WALLET_2));
         given(walletService.getAll()).willReturn(listOfWalletsDTO);
-        given(walletService.findByName(walletNameSearched)).willReturn(foundedWalletsDTO);
+        given(walletService.findAllByName(walletNameSearched)).willReturn(foundedWalletsDTO);
 
         // then
         mockMvc.perform(get("/api/wallet/find/{name}", walletNameSearched)
```
Comment (DawidStuzynski) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/WalletControllerTest.java:442]: API paths should not contains verbs,  `/api/wallet/{name}`  I think would be better.

**Review by DawidStuzynski — CHANGES_REQUESTED** [review]
Request changes.

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/repository/WalletRepository.java (hunk 1 above)
Comment (Iwona007) [reply src/main/java/pl/byczazagroda/trackexpensesappbackend/repository/WalletRepository.java:16]: Yes, you have right, but it was my suggestion to write with SQL to practse this query

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/WalletControllerTest.java (hunk 4 above)
Comment (Iwona007) [reply src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/WalletControllerTest.java:442]: you can change for example: /api/wallet/list/{name}

**Review by Iwona007 — COMMENTED** [review]
comment added

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/WalletControllerTest.java (hunk 4 above)
Comment (dawciobiel) [reply src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/WalletControllerTest.java:442]: > you can change for example: /api/wallet/list/{name}

```markdown
2.8 Filter Operations

While building RESTful APIs, we will come across multiple requirements where we need collection of resources with

    Sorting
    Filtering our certain results from the collection.
    Limited results based on certain conditions 

Do not create new API for these requirements.Use query parameters to achieve all these results.

http://www.javadevjournal.com/products?catalog=Online
http://www.javadevjournal.com/products?region=USA&brand=dummy&sort=new

```

**source**
https://www.javadevjournal.com/spring/rest-resource-naming/

**Review by dawciobiel — CHANGES_REQUESTED** [review]
Changes required

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/service/WalletServiceImpl.java (hunk 2 above)
Comment (Krisu93) [reply src/main/java/pl/byczazagroda/trackexpensesappbackend/service/WalletServiceImpl.java:87]: In line 57 (methid getAll()) is same

**Review by Iwona007 — APPROVED** [review]

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/WalletControllerTest.java [hunk 5]
```diff
@@ -436,10 +436,10 @@ void shouldReturnListOfAllWalletsByName() throws Exception {
         List<WalletDTO> listOfWalletsDTO = createListOfWalletsDTO();
         List<WalletDTO> foundedWalletsDTO = List.of(new WalletDTO(ID_OF_WALLET_2, NAME_OF_WALLET_2, CREATION_DATE_OF_WALLET_2));
         given(walletService.getAll()).willReturn(listOfWalletsDTO);
-        given(walletService.findByName(walletNameSearched)).willReturn(foundedWalletsDTO);
+        given(walletService.findAllByName(walletNameSearched)).willReturn(foundedWalletsDTO);
 
         // then
-        mockMvc.perform(get("/api/wallet/find/{name}", walletNameSearched)
+        mockMvc.perform(get("/api/wallet/list/{name}", walletNameSearched)
```
Comment (odzio33) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/WalletControllerTest.java:442]: We should retire "list" from our path. In my opinion the correct endpoint shoud be /api/wallet/{name}

Comment (mirekgab) [reply src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/WalletControllerTest.java:442]: If we do it this way, we'll get an exception: Ambiguous handler methods mapped for "/api/wallet/{name}". This exception is related to the findById(java.lang.Long) and findByName(java.lang.String) methods.

**Review by dawciobiel — APPROVED** [review]

**Pull request merged**